src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,49 @@
 package core.basesyntax;
 
-public class SalaryInfo {
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public class SalaryInfo extends CheckNotNull {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        checkNotNull(names, data, dateFrom, dateTo);
+        String[] fullInformation = new String[names.length];
+        StringBuilder builder;
+        StringBuilder finalBuilder = new StringBuilder();
+        SimpleDateFormat format = new SimpleDateFormat("dd.MM.yyyy");
+        short builderCounter = 0;
+        for (String name : names) {
+            int sum = 0;
+            builder = new StringBuilder();
+            for (String note : data) {
+                if (!note.isEmpty()) {
+                    try {
+                        String[] dataSeparated = note.split(" ");
+                        Date parsedNoteDate = format.parse(dataSeparated[0]);
```

Comment:
It is recommended not to use `Date` class for date representation. Instead, please use `LocalDate`.

[Code recommendations](https://mate-academy.github.io/jv-program-common-mistakes/java-core/exceptions/salary-info#use-localdate-instead-of-date)